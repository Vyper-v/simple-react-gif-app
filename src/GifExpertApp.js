import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifLayout from "./components/GifLayout";

function GifExpertApp({ defaultCategories = [] }) {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <div className=" container mx-auto divide-y divide-neutral-500 divide-dashed">
        <h2 className="text-4xl mb-8 leading-normal">GifExpertApp</h2>
        <div className="flex justify-between px-2 py-4">
          <AddCategory setCategories={setCategories} />
        </div>
      </div>
      <div className="container mx-auto space-y-2">
        {categories.map((category) => {
          return (
            <div key={category}>
              <a
                href={`#${category}`}
                className="text-2xl m-2 inline-block text-lime-700 hover:text-lime-600"
              >
                # {category}
              </a>
              <GifLayout id={category} category={category} />
            </div>
          );
        })}
      </div>
    </div>
  );
}

export default GifExpertApp;
