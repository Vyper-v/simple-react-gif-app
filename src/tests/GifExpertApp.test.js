import GifExpertApp from "../GifExpertApp";
import { shallow } from "enzyme";
describe("app test", () => {
  test("should be rendered", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should show categories list", () => {
    const categories = ["Dragon Ball Z", "One Piece"];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifLayout").length).toBe(categories.length); 
  });
});
