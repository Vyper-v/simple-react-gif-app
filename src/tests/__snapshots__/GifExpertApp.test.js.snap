// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test should be rendered 1`] = `
<div>
  <div
    className=" container mx-auto divide-y divide-neutral-500 divide-dashed"
  >
    <h2
      className="text-4xl mb-8 leading-normal"
    >
      GifExpertApp
    </h2>
    <div
      className="flex justify-between px-2 py-4"
    >
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </div>
  <div
    className="container mx-auto space-y-2"
  />
</div>
`;

exports[`app test should show categories list 1`] = `
<div>
  <div
    className=" container mx-auto divide-y divide-neutral-500 divide-dashed"
  >
    <h2
      className="text-4xl mb-8 leading-normal"
    >
      GifExpertApp
    </h2>
    <div
      className="flex justify-between px-2 py-4"
    >
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </div>
  <div
    className="container mx-auto space-y-2"
  >
    <div
      key="Dragon Ball Z"
    >
      <a
        className="text-2xl m-2 inline-block text-lime-700 hover:text-lime-600"
        href="#Dragon Ball Z"
      >
        # 
        Dragon Ball Z
      </a>
      <GifLayout
        category="Dragon Ball Z"
        id="Dragon Ball Z"
      />
    </div>
    <div
      key="One Piece"
    >
      <a
        className="text-2xl m-2 inline-block text-lime-700 hover:text-lime-600"
        href="#One Piece"
      >
        # 
        One Piece
      </a>
      <GifLayout
        category="One Piece"
        id="One Piece"
      />
    </div>
  </div>
</div>
`;
